commit 7023207cfdb6d5863bdcc9c29bff9918846dddc6
Author: Tom Lane <tgl@sss.pgh.pa.us>
Date:   Fri Mar 24 19:31:19 2023 -0400

    Doc: fix examples for pg_input_error_info().
    
    These were causing "contents ... exceed the available area"
    warnings in PDF builds, and also didn't quite follow our markup
    conventions for function examples.  To fix the overwidth
    problem, reduce the number of fields shown in one example,
    and also insert &zwsp; to let the header line be broken in
    a reasonable place.
    
    Discussion: https://postgr.es/m/20230324194701.dqkzcdtlcikseo22@awork3.anarazel.de

diff --git a/doc/src/sgml/func.sgml b/doc/src/sgml/func.sgml
index 0987eb805a9..974d7be8c2f 100644
--- a/doc/src/sgml/func.sgml
+++ b/doc/src/sgml/func.sgml
@@ -24900,32 +24900,37 @@ SELECT collation for ('foo' COLLATE "de_DE");
         <parameter>sql_error_code</parameter> <type>text</type> )
        </para>
        <para>
         Tests whether the given <parameter>string</parameter> is valid
         input for the specified data type; if not, return the details of
-        the error would have been thrown.  If the input is valid, the
+        the error that would have been thrown.  If the input is valid, the
         results are NULL.  The inputs are the same as
         for <function>pg_input_is_valid</function>.
        </para>
        <para>
         This function will only work as desired if the data type's input
         function has been updated to report invalid input as
         a <quote>soft</quote> error.  Otherwise, invalid input will abort
         the transaction, just as if the string had been cast to the type
         directly.
-        </para>
-        <para>
+       </para>
+       <para>
+        <literal>select * from pg_input_error_info('42000000000', 'integer')</literal>
+        <returnvalue></returnvalue>
 <programlisting>
-SELECT * FROM pg_input_error_info('42000000000', 'integer');
                        message                        | detail | hint | sql_error_code
 ------------------------------------------------------+--------+------+----------------
  value "42000000000" is out of range for type integer |        |      | 22003
-
-SELECT * FROM pg_input_error_info('1234.567', 'numeric(7,4)');
-        message         |                                      detail                                       | hint | sql_error_code
-------------------------+-----------------------------------------------------------------------------------+------+----------------
- numeric field overflow | A field with precision 7, scale 4 must round to an absolute value less than 10^3. |      | 22003
+</programlisting>
+       </para>
+       <para>
+        <literal>select message, detail from pg_input_error_info('1234.567', 'numeric(7,4)')</literal>
+        <returnvalue></returnvalue>
+<programlisting>
+        message         |                                      detail
+------------------------+----------------------------------&zwsp;-------------------------------------------------
+ numeric field overflow | A field with precision 7, scale 4 must round to an absolute value less than 10^3.
 </programlisting>
        </para></entry>
       </row>
      </tbody>
     </tgroup>
